File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp = head;
        int ttl = 0;

        while (temp != null) {
            ttl++;
            temp = temp.next;
        }

        if (n == ttl) return head.next;

        int cnt = 0;
        temp = head;

        while (cnt < ttl - n - 1) {
            temp = temp.next;
            cnt++;
        }

        temp.next = temp.next.next;

        return head;
    }
}

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        
        ListNode* slow = head;
        ListNode* fast = head;
        
        for(int i = 1; i<= n; i++){
            fast = fast->next;
        }
        
        if(fast == NULL){
            return head->next;
        }
        
        while(fast != NULL && fast->next != NULL){
            fast = fast->next;
            slow = slow->next;
        }
        
        slow->next = slow->next->next;
        
        return head;
    }
};